src/scipp/core/binning.py

```diff
@@ -167,20 +167,20 @@ def _parse_coords_arg(x, name, arg):
     if isinstance(arg, Variable) and name in arg.dims:
         return arg
     coord = _get_coord(x, name)
+    start = coord.min()
+    if not isinstance(x, Variable) and (name in x.meta) and x.meta.is_edges(name, name):
+        stop = coord.max()  # existing bin-edges, do not extend
+    else:
+        stop = _upper_bound(coord)
```

Comment:
The type hint in `_upper_bound` isn't correct here. If `x` is a data array but does not have a bin edge coord in `dim=name`, we still hit this branch and call `_upper_bound`.

Reply:
Never mind, misread the code